configs/configload/copy_dir.go

```diff
@@ -59,6 +60,25 @@ func copyDir(dst, src string) error {
 			return nil
 		}
 
+		// info.IsDir() returns false for symlinks, even if the symlink is a
+		// directory.
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			_, err := ioutil.ReadDir(path)
+			if err == nil {
+				// it was a directory ALL ALONG!!!
+				if path == filepath.Join(src, dst) {
+					// dst is in src; don't walk it.
+					return nil
+				}
+
+				if err := os.MkdirAll(dstPath, 0755); err != nil {
```

Comment:
I'm a little unclear on the control flow here... after we've made this directory, does `filepath.Walk` then visit the contents of the directory that the symlink refers to? If it doesn't, then wouldn't we just end up with an empty directory here?

I wonder if we need to recursively call `copyDir` here to also copy the _contents_ of the directory, rather than just making the directory.

Reply:
I must admit I copied this logic from the preceding section (`if info.IsDir() {`). I _thought_ that was the behavior, but on closer inspection that doesn't seem to be the case. I will keep poking at it and extend the test coverage to include a symlinked module, too.